rfc/0018-certificate-handling.md

```diff
@@ -82,174 +86,195 @@ webhooks:
+> Server. The new certificate is still signed by the same CA, hence nothing has
 > to be changed.
 
 ### Internal CA changes
 
-The internal CA managed by Kubewarden might be recreated (for example, to avoid its expiration). This is the
-most disruptive event that can happen, especially once some policies are deployed.
+The internal CA managed by Kubewarden might be recreated (for example, to avoid
+its expiration). This is the most disruptive event that can happen, especially
+once some policies are deployed.
 
-Once the new CA is generated, the following actions have to be performed:
+The following actions have to be performed to renew the internal CA:
 
-- Generate a new CA bundle that contains the previous internal CA and the new one
-- For each webhook configuration managed by Kubewarden (meaning all the policies deployed, plus the kubewarden-controller):
-  - Update the `clientConfig.caBundle`: ensure it contains the CA bundle created during the previous step
+- Generates a new CA certificate
+- Updates the secret storing the CA certificate with the new CA and the old CA
+- Generate a new CA bundle that contains the previous internal CA and the new
+  one
+- For each webhook configuration managed by Kubewarden (meaning all the
+  policies deployed, plus the `kubewarden-controller`):
+  - Update the `clientConfig.caBundle`: ensure it contains the CA bundle
+    created during the previous step
 - For each `PolicyServer` defined:
   - Generate a new certificate, sign it with the new CA
-  - Force a rollout of the PolicyServer Deployment
-- Kubewarden Controller:
+  - Update the secret storing the policy server certificate
+- For the `kubewarden-controller`:
   - Generate a new certificate, sign it with the new CA
-  - Restart the controller to ensure its HTTPS endpoint uses the new certificate
+  - Update the secret storing the controller certificate
 
-Once all the rollout of the new Policy Server instances is done, and the kubewarden controller is using the new certificate:
+All certificates generated uses the ECDSA P-256 curve algorithm for the keys.
+Once the old CA is invalid, and the kubewarden controller is using the new
+certificate:
 
-- For each webhook configuration managed by Kubewarden (meaning all the policies deployed, plus the kubewarden-controller):
-  - Update the `clientConfig.caBundle`: ensure it contains only the certitificate of the new CA
+- For each webhook configuration managed by Kubewarden (meaning all the
+  policies deployed, plus the `kubewarden-controller`):
+  - Update the `clientConfig.caBundle`: ensure it contains only the
+    certitificate of the new CA
 
-By respecing these steps we can avoid communication failures between the Kubernetes API server and the webhooks.
+> **Note:** there's no need to restart the Policy Server Pod. Both Policy
+> Server and the `kubewarden-controller` are able to detect changes in the
+> certificate file and reload it on runtime.
```

Comment:
```suggestion
The root CA rotation is initiated 60 days before the expiration date.
This gives plenty of time to ensure all the Policy Server instances and the
`kubewarden-controller` pods are using a certificate that has been issued
by the new root CA.

Once the old CA is invalid:
- Remove the old root CA from the CA bundle
- For each webhook configuration managed by Kubewarden (meaning all the
  policies deployed, plus the validating and mutating webhooks
  defined by the Kubewarden Controller):
  - Update the `clientConfig.caBundle`: use the freshly computed bundle
  
> **Note:** there's no need to restart the Policy Server/kubewarden-controller pods.
> By that time (60 days passed since the new root CA and their certificates were
> recreated), they are all using a certificate issued by the new root CA.
```